pkg/build/cmd/grafanacom.go

```diff
@@ -0,0 +1,310 @@
+	case config.EditionOSS:
+		pth = "oss"
+	case config.EditionEnterprise:
+		pth = "enterprise"
+		sfx = EnterpriseSfx
+	default:
+		return fmt.Errorf("unrecognized edition %q", cfg.Edition)
+	}
+
+	switch cfg.ReleaseMode.Mode {
+	case config.MainMode, config.CustomMode, config.CronjobMode:
+		pth = path.Join(pth, MainFolder)
+	default:
+		pth = path.Join(pth, ReleaseFolder)
+	}
+
+	product := fmt.Sprintf("grafana%s", sfx)
+	pth = path.Join(pth, product)
+	baseArchiveURL := fmt.Sprintf("https://dl.grafana.com/%s", pth)
+
+	var builds []buildRepr
+	for _, ba := range ArtifactConfigs {
+		u := ba.GetURL(baseArchiveURL, cfg)
+
+		sha256, err := getSHA256(u)
+		if err != nil {
+			return err
+		}
+
+		builds = append(builds, buildRepr{
+			OS:     ba.Os,
+			URL:    u,
+			SHA256: string(sha256),
+			Arch:   ba.Arch,
+		})
+	}
+
+	r := releaseRepr{
+		Version:     cfg.Version,
+		ReleaseDate: time.Now().UTC(),
+		Builds:      builds,
+		Stable:      cfg.ReleaseMode.Mode == config.TagMode,
+		Beta:        cfg.ReleaseMode.IsBeta,
+		Nightly:     cfg.ReleaseMode.Mode == config.CronjobMode,
+	}
+	if cfg.ReleaseMode.Mode == config.TagMode || r.Beta {
+		r.WhatsNewURL = cfg.WhatsNewURL
+		r.ReleaseNotesURL = cfg.ReleaseNotesURL
+	}
+
+	if err := postRequest(cfg, "versions", r, fmt.Sprintf("create release %s", r.Version)); err != nil {
+		return err
+	}
+
+	if err := postRequest(cfg, fmt.Sprintf("versions/%s", cfg.Version), r,
+		fmt.Sprintf("update release %s", cfg.Version)); err != nil {
+		return err
+	}
+
+	for _, b := range r.Builds {
+		if err := postRequest(cfg, fmt.Sprintf("versions/%s/packages", cfg.Version), b,
+			fmt.Sprintf("create build %s %s", b.OS, b.Arch)); err != nil {
+			return err
+		}
+		if err := postRequest(cfg, fmt.Sprintf("versions/%s/packages/%s/%s", cfg.Version, b.Arch, b.OS), b,
+			fmt.Sprintf("update build %s %s", b.OS, b.Arch)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getSHA256(u string) ([]byte, error) {
+	shaURL := fmt.Sprintf("%s.sha256", u)
+	resp, err := http.Get(shaURL)
+	if err != nil {
+		return nil, err
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			return
+		}
+	}(resp.Body)
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return nil, fmt.Errorf("failed downloading %s: %s", u, resp.Status)
+	}
+
+	var sha256 []byte
+	if err := json.NewDecoder(resp.Body).Decode(&sha256); err != nil {
+		return nil, err
+	}
+	return sha256, nil
+}
+
+func postRequest(cfg PublishConfig, pth string, obj interface{}, descr string) error {
+	var sfx string
+	switch cfg.Edition {
+	case config.EditionOSS:
+	case config.EditionEnterprise:
+		sfx = EnterpriseSfx
+	default:
+		return fmt.Errorf("unrecognized edition %q", cfg.Edition)
+	}
+	product := fmt.Sprintf("grafana%s", sfx)
+
+	jsonB, err := json.Marshal(obj)
+	if err != nil {
+		return fmt.Errorf("failed to JSON encode release: %w", err)
+	}
+
+	u := fmt.Sprintf("https://grafana.com/api/%s/%s", product, pth)
```

Comment:
It could be good to protect against path travel: if `product` or`pth` contains `../other_endpoint/whatever`, another grafana.com endpoint could be called (success will depend on permissions granted to the API key)


Slide 7 of [this talk](https://docs.google.com/presentation/d/1S6VTF_mvXRC_du3bGQqUj3R3GyOZiYcY7cPe3wn80Cg/edit#slide=id.g1596b83a5d9_0_76) contains a good recommendation for fixing.
Ideally all the url built to call grafana com api should follow that